iguana.corecontroller/src/main/java/org/aksw/iguana/cc/utils/IndexedLineReader.java

```diff
@@ -5,67 +5,91 @@
 import java.io.IOException;
 import java.io.RandomAccessFile;
 import java.nio.charset.StandardCharsets;
-import java.nio.file.Files;
-import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
 
 /**
  * This class creates an object, that indexes the starting positions of the lines inside a file for faster access. <br/>
  * The indexing happens on either:
  * <ul>
  *     <li>the file's line endings, or</li>
- *     <li>a custom line separator</li>
+ *     <li>a custom separator</li>
  * </ul>
  * This class does not index blank lines.
  */
 public class IndexedLineReader {
 
     /** This list stores the indices of the bytes, at which a new line starts. (Stores Integers, so it only supports
      *  files that are roughly 2GB big.)*/
-    private ArrayList<Integer> indices;
+    private ArrayList<Integer[]> indices;
```

Comment:
- [ ] Create an issue to track the limitation to 2GB. We will want to extend it at a later point. 
- [ ] The internal array should store offset + length instead of start offset + end offset (see further down)
- [ ] The internal array must be documented.

Reply:
I have changed the type, that the list stores, to Long